### DISCUSSION TITLE
recover disks from using hdparm

### DISCUSSION THREAD

**Discussion by Darkseid51** [1000001101000/Debian_on_Buffalo#141] [Q&A]
Hi,

Not strictly Debian_on_Buffalo question, but side effect of something going wrong...

So I had Debian 10 installed successfully on my LS-QVL.
I had a pair of 2Tb and pair of 3Tb drives mounted as XFS raid 1 arrays created with the Buffalo stock firmware.

Other than some issues with fancontrol all good (rpm vs pwm reversed so 5000rpm was fan stopped, so simply set at a mid value as couldn't get fancontrol to accept mapping values opposite way around)

Unfortunately on power disconnect / reconnect none of the HDD's spin up in the NAS :(

I *presume* this is a side effect of using hdparm to power disks down after a few hours of no use.
Seemed to be working as expected - tested on one of each type, would stop on inactivity and spin up on use ... until physical power disconnect of the NAS.

I've tried a couple of SATA/USB enclosures on Linux and Windows however the disks fail to spin up.

Buffalo firmware emergency recovery app throws 'format failed'.

Both hdparm and scsi-spin seem to need a /dev/sdX reference as a parameter and I don't get that far...

On connection to a Mint 20.3 desktop build:

[ 7687.314672] usb 2-4: new SuperSpeed Gen 1 USB device number 15 using xhci_hcd
[ 7687.335349] usb 2-4: New USB device found, idVendor=1058, idProduct=1230, bcdDevice=10.50
[ 7687.335353] usb 2-4: New USB device strings: Mfr=2, Product=3, SerialNumber=1
[ 7687.335355] usb 2-4: Product: My Book 1230
[ 7687.335358] usb 2-4: Manufacturer: Western Digital
[ 7687.335359] usb 2-4: SerialNumber: 574343344530303730363837
[ 7687.336686] usb-storage 2-4:1.0: USB Mass Storage device detected
[ 7687.336955] scsi host4: usb-storage 2-4:1.0
[ 7688.363269] scsi 4:0:0:0: Direct-Access     WD       My Book 1230     1050 PQ: 0 ANSI: 6
[ 7688.363644] scsi 4:0:0:1: Enclosure         WD       SES Device       1050 PQ: 0 ANSI: 6
[ 7688.364851] sd 4:0:0:0: Attached scsi generic sg0 type 0
[ 7688.365082] ses 4:0:0:1: Attached Enclosure device
[ 7688.365257] ses 4:0:0:1: Attached scsi generic sg1 type 13
[ 7725.098629] usb 2-4: Disable of device-initiated U1 failed.
[ 7730.218682] usb 2-4: Disable of device-initiated U2 failed.
[ 7730.347031] usb 2-4: reset SuperSpeed Gen 1 USB device number 15 using xhci_hcd
[ 7730.368524] sd 4:0:0:0: [sda] 976746240 4096-byte logical blocks: (4.00 TB/3.64 TiB)
[ 7730.368930] ses 4:0:0:1: Failed to get diagnostic page 0x1
[ 7730.368938] ses 4:0:0:1: Failed to bind enclosure -19
[ 7730.369172] sd 4:0:0:0: [sda] Write Protect is off
[ 7730.369176] sd 4:0:0:0: [sda] Mode Sense: 53 00 10 08
[ 7730.369420] sd 4:0:0:0: [sda] No Caching mode page found
[ 7730.369424] sd 4:0:0:0: [sda] Assuming drive cache: write through
[ 7745.381743] sd 4:0:0:0: [sda] Spinning up disk...
[ 7746.410538] ........not responding...

start loop here every 30 seconds:

[ 7868.410702] sd 4:0:0:0: [sda] tag#0 FAILED Result: hostbyte=DID_OK driverbyte=DRIVER_SENSE
[ 7868.410707] sd 4:0:0:0: [sda] tag#0 Sense Key : Not Ready [current] 
[ 7868.410710] sd 4:0:0:0: [sda] tag#0 Add. Sense: No additional sense information
[ 7868.410713] sd 4:0:0:0: [sda] tag#0 CDB: Read(10) 28 00 00 00 00 00 00 00 01 00
[ 7868.410718] blk_update_request: I/O error, dev sda, sector 0 op 0x0:(READ) flags 0x0 phys_seg 1 prio class 0
[ 7868.410725] Buffer I/O error on dev sda, logical block 0, async page read

The disk above is a Western Digital 3.0TB 'WD RED NAS Hard Drive' WD30EFRX

The others are Samsung HD204UI

Any ideas?

I'm hoping I've missed something trivial that allows me to send a spin up command...

Re,
Chris

**Comment by Darkseid51** [comment]
So the good news is when connected on a SATA cable to my desktop PC the disk spins up and partitions are there.
Even passes WD diagnostics test, but no options to change any disk settings (other than erase).

The bad news inside the LS-QVL and via two different USB3 SATA adaptors the drive stays spun down on Linux NUC and PC.
Pity the NUC doesn't have SATA otherwise could have tried a hdparm disable.

Any ideas? Google not much help with disk no spinning up (even with hdparm qualifier) as brings up every user's failed HDD or power problems

**Comment by 1000001101000** [comment]
Good to hear from you!

I don't have any experience with messing with hdparm stuff and haven't heard of this issue before so mostly only have general thoughts/advice. 

Could you post the exact commands/parameters you sent to the disks with hdparm? I could try and reproduce it with my LS-QVL and take a look at some documentation. 

If your PC is able to talk to it reliably I think your best bet in the short term would be to try booting your PC into a Linux environment and see if the disks are still recognized, then:
1.  Backup your data if you haven't already
2.  boot into Linux on the PC and use hdparm to undo whatever setting change you made and see if that makes it behave normally again. 

I don't want to jump to any conclusions since that's an easy thing to do with this type of issue. I could imagine there being a difference with how your PC interacts with the drive via direct connection to the SATA port vs through a USB->SATA chipset or even the SATA multiplexer in the LS-QVL. It might be worth adding that info to your searches to see if that helps or not. I would think it would be a well known issue if it was that simple. I've seen some interesting things connecting SSDs to ancient systems via IDE->SATA adapters where Linux detects that the drive supports TRIM but the adapter doesn't (likely the chip predates TRIM) leading to errors/etc. In that case it was as simple as just disabling TRIM at the system level.

**Reply by Darkseid51** [reply]
Hi again,

Command was as follows, with varying short test times for one disk then around 2 hours I think applied to each:

hdparm -S 240 /dev/sda

/dev/sda:
setting standby to 240 (20 minutes)

I was then distracted by fancontrol (turns off fan when max temp reached) and left copying files at ~48MB/sec for a day.
Powered off to move out of the way and then disks didn't spin up thereafter.

-=-

The disks spin up when connected to SATA on a PC, though they don't start when the rest do.
Normally I hear spin up of an existing disk, BIOS POST, delay for function key press on any hardware changes then OS boot.

With one of these plugged in hear spin up of existing disk, then several seconds delay, then ex-NAS disk spin up, then BIOS POST.

Booting to Linux on the PC I can access and retrieve disk data and properties, however the undo command whilst accepted has no effect.

hdparm -S 0 /dev/sda

-=-

hdparm -I /dev/sda

/dev/sda:

ATA device, with non-removable media
powers-up in standby; SET FEATURES subcmd spins-up.
    Model Number:   	SAMSUNG HD204UI                    	 
    Serial Number:  	S2H7J1CB209083 	 
    Firmware Revision:  1AQ10001
    Transport:      	Serial, ATA8-AST, SATA 1.0a, SATA II Extensions, SATA Rev 2.5, SATA Rev 2.6
Standards:
    Used: unknown (minor revision code 0x0028)
    Supported: 8 7 6 5
    Likely used: 8
Configuration:
    Logical   	 max    current
    cylinders    16383    16383
    heads   	 16    16
    sectors/track    63    63
    --
    CHS current addressable sectors:	16514064
    LBA	user addressable sectors:   268435455
    LBA48  user addressable sectors:  3907029168
    Logical  Sector size:               	512 bytes
    Physical Sector size:               	512 bytes
    device size with M = 1024*1024: 	1907729 MBytes
    device size with M = 1000*1000: 	2000398 MBytes (2000 GB)
    cache/buffer size  = unknown
    Form Factor: 3.5 inch
    Nominal Media Rotation Rate: 5400
Capabilities:
    LBA, IORDY(can be disabled)
    Queue depth: 32
    Standby timer values: spec'd by Standard, no device specific minimum
    R/W multiple sector transfer: Max = 16    Current = 16
    Advanced power management level: 254
    Recommended acoustic management value: 254, current value: 0
    DMA: mdma0 mdma1 mdma2 udma0 udma1 udma2 udma3 udma4 udma5 *udma6
     	Cycle time: min=120ns recommended=120ns
    PIO: pio0 pio1 pio2 pio3 pio4
     	Cycle time: no flow control=120ns  IORDY flow control=120ns
Commands/features:
    Enabled    Supported:
   	*    SMART feature set
   		 Security Mode feature set
   	*    Power Management feature set
   	*    Write cache
   	*    Look-ahead
   	*    Host Protected Area feature set
   	*    WRITE_BUFFER command
   	*    READ_BUFFER command
   	*    NOP cmd
   	*    DOWNLOAD_MICROCODE
   	*    Advanced Power Management feature set
   	*    Power-Up In Standby feature set
   	*    SET_FEATURES required to spinup after power up
   		 SET_MAX security extension
   		 Automatic Acoustic Management feature set
   	*    48-bit Address feature set
   	*    Device Configuration Overlay feature set
   	*    Mandatory FLUSH_CACHE
   	*    FLUSH_CACHE_EXT
   	*    SMART error logging
   	*    SMART self-test
   	*    General Purpose Logging feature set
   	*    64-bit World wide name
   	*    WRITE_UNCORRECTABLE_EXT command
   	*    {READ,WRITE}_DMA_EXT_GPL commands
   	*    Segmented DOWNLOAD_MICROCODE
   	*    Gen1 signaling speed (1.5Gb/s)
   	*    Gen2 signaling speed (3.0Gb/s)
   	*    Native Command Queueing (NCQ)
   	*    Host-initiated interface power management
   	*    Phy event counters
   	*    NCQ priority information
   	*    DMA Setup Auto-Activate optimization
   		 Device-initiated interface power management
   	*    Software settings preservation
   	*    SMART Command Transport (SCT) feature set
   	*    SCT Read/Write Long (AC1), obsolete
   	*    SCT Write Same (AC2)
   	*    SCT Error Recovery Control (AC3)
   	*    SCT Features Control (AC4)
   	*    SCT Data Tables (AC5)
Security:
    Master password revision code = 65534
   	 supported
    not    enabled
    not    locked
   	 frozen
    not    expired: security count
   	 supported: enhanced erase
    328min for SECURITY ERASE UNIT. 328min for ENHANCED SECURITY ERASE UNIT.
Logical Unit WWN Device Identifier: 50024e9004a82a5c
    NAA   	 : 5
    IEEE OUI    : 0024e9
    Unique ID    : 004a82a5c
Checksum: correct

-=-

I've also tried a factory firmware restore

hdparm --dco-restore /dev/hda 

(then again with the yes please destroy my disk confirmation parameter)
